mailbox/plugin/spamassassin/src/main/java/org/apache/james/mailbox/spamassassin/SpamAssassinConfiguration.java

```diff
@@ -25,13 +25,25 @@
 import org.apache.james.util.Host;
 
 import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
 
 public class SpamAssassinConfiguration {
 
     private final Optional<Host> host;
+    private final Optional<Integer> parallelism;
 
-    public SpamAssassinConfiguration(Optional<Host> host) {
+    public static SpamAssassinConfiguration disabled() {
+        return new SpamAssassinConfiguration(Optional.empty(), Optional.empty());
+    }
+
+    public static SpamAssassinConfiguration enabled(Host host, Integer parallelism) {
+        Preconditions.checkArgument(parallelism > 0, "Parallelism should be at least one");
```

Comment:
could we rather speak about concurrency rather than parallelism?

Reply:
I think it's more parallelism related here.

Reply:
Could you elaborate?

Reply:
concurrency is more independence of processings, while here we want to have a parallel execution (a parallel send of the requests).

Reply:
what difference do you make between "running parallel queries" and "interleave concurrent queries"?
My interpretation is that parallel is a way of doing things _at the same time_ (== concurrently) but time-sharing is also. So it would be an implementation detail because what the user sees is "how many things are going on at the same time", parallel vs interleaved has no value from him.